packages/hub/tasks/scheduled-payment-on-chain-execution-waiter.ts

```diff
@@ -51,64 +51,49 @@ export default class ScheduledPaymentOnChainExecutionWaiter {
         }
       }
     } catch (error: any) {
-      // waitUntilTransactionMined will return "Transaction took too long to complete" in case it wasn't mined in 60 minutes.
-      // In this case we want to restart the task and wait some more. We could throw an error here so that the worker would restart the task, but
-      // due to the exponential-backoff it could take some time before the waiting can start again. That's why we just spawn a new task directly.
-      if (error.message.includes('took too long')) {
-        if (isBefore(paymentAttempt.startedAt!, subDays(nowUtc(), 1))) {
-          await prisma.scheduledPaymentAttempt.update({
-            data: {
-              status: 'failed',
-              failureReason: "Waited for more than 1 day for the transaction to be mined, but it wasn't",
-              endedAt: nowUtc(),
-            },
-            where: {
-              id: paymentAttempt.id,
-            },
-          });
-        } else {
-          await this.workerClient.addJob('scheduled-payment-on-chain-execution-waiter', {
-            scheduledPaymentAttemptId: paymentAttempt.id,
-          });
-        }
-
-        await prisma.scheduledPayment.update({
-          where: { id: scheduledPayment.id },
-          data: {
-            nextRetryAttemptAt: this.scheduledPaymentFetcher.calculateNextRetryAttemptDate(scheduledPayment),
-          },
-        });
-
-        return;
-      }
-
       // Known errors are:
       // - Transaction with hash "${txnHash}" was reverted
       // - UnknownHash: payment details generate unregistered spHash
       // - InvalidPeriod: payment executed outside of valid date or period
       // - ExceedMaxGasPrice: gasPrice must be lower than or equal maxGasPrice
       // - PaymentExecutionFailed: safe balance is not enough to make payments and pay fees
       // - OutOfGas: executionGas to low to execute scheduled payment
-      //
-      // In these cases we want to mark the payment attempt as failed and set the next retry attempt date
-      await prisma.scheduledPaymentAttempt.update({
-        data: {
-          status: 'failed',
-          failureReason: error.message,
-          endedAt: nowUtc(),
-        },
-        where: {
-          id: paymentAttempt.id,
-        },
-      });
-
-      await prisma.scheduledPayment.update({
-        where: { id: scheduledPayment.id },
-        data: {
-          nextRetryAttemptAt: this.scheduledPaymentFetcher.calculateNextRetryAttemptDate(scheduledPayment),
-        },
-      });
+      let knownErrors = [
+        'was reverted',
+        'UnknownHash',
+        'InvalidPeriod',
+        'ExceedMaxGasPrice',
+        'PaymentExecutionFailed',
+        'OutOfGas',
+      ];
+      let isKnownError = knownErrors.find((knownError) => error.message.includes(knownError));
+      let isWaitTooLongError =
```

Comment:
I think it should be `let isWaitTooLongError = error.message.includes('took too long')` (without the `startedAt` check) to not mix unrelated things. And then you can have another check for `startedAt` later

Reply:
I update the naming variable to `isWaitTooLong` instead. So that variable tells us whether we wait too long for that payment.

Reply:
I think it's still a misleading that way. We have 2 different validations:

1. The allowed time that the waiter process to wait for a verified transaction in the waiter has expired
2. The time that we allow the transaction to be pending or to appear on chain has expired

I suggest the following naming to make things clearer:

`let waiterTimedOut = error.message.includes('took too long');`
`let waitingExpired = isBefore(paymentAttempt.startedAt!, subDays(nowUtc(), 1));`